dbld/images/helpers/dependencies.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+set -e
+set -x
+
+function install_criterion() {
+    CRITERION_VERSION=2.3.2
+    download_target "https://github.com/Snaipe/Criterion/releases/download/v${CRITERION_VERSION}/criterion-v${CRITERION_VERSION}-linux-x86_64.tar.bz2" /tmp/criterion.tar.bz2
+    tar xvjf /tmp/criterion.tar.bz2 --strip 1 -C /usr
+    rm -rf /tmp/criterion.tar.bz2
+}
+
+function install_gosu() {
+    GOSU_VERSION=1.10
+    ARCHITECTURE=$1
+    cat /helpers/gosu.pubkey | gpg --import
+    download_target "https://github.com/tianon/gosu/releases/download/${GOSU_VERSION}/gosu-${ARCHITECTURE}" /usr/local/bin/gosu
+    download_target "https://github.com/tianon/gosu/releases/download/${GOSU_VERSION}/gosu-${ARCHITECTURE}.asc" /usr/local/bin/gosu.asc
+    gpg --verify /usr/local/bin/gosu.asc
+    rm /usr/local/bin/gosu.asc
+    chmod +x /usr/local/bin/gosu
+}
+
+function install_gradle {
+    GRADLE_VERSION=4.10
+    download_target "https://services.gradle.org/distributions/gradle-${GRADLE_VERSION}-bin.zip" /tmp/gradle.zip
+    mkdir -p /opt/gradle
+    unzip -d /opt/gradle /tmp/gradle.zip
+    rm -rf /tmp/gradle.zip
+    ln -s /opt/gradle/gradle-${GRADLE_VERSION}/bin/gradle /usr/bin/gradle
+}
+
+function download_target() {
+    target=$1
+    output=$2
+    wget --no-check-certificate $target --output-document=$output
+}
+
+function install_apt_packages {
+    OS_PLATFORM=$1
+    apt-get update -qq -o Acquire::CompressionTypes::Order::=gz
+    grep ${OS_PLATFORM} /packages/packages.txt | cut -d"[" -f1 | xargs apt-get install --no-install-recommends --yes
```

Comment:
optional: to avoid unintentional matches I would filter out comments before matching to OS_PLATFORM.
optional2: add the surrounding '[' brackets to the expression to make sure no package name will conflict with any platform.